### ISSUE TITLE
Install with Ruby 2.5.1, RubyGems 2.7.7 prompts about executable "bundle" conflict

### ISSUE THREAD

**Issue by knutsenm** [thoughtbot/laptop#543]
The `mac` script prompted me thusly:

```
RubyGems installed the following executables:
        /Users/knutsen/.asdf/installs/ruby/2.5.1/bin/gem
        /Users/knutsen/.asdf/installs/ruby/2.5.1/bin/bundle

Ruby Interactive (ri) documentation was installed. ri is kind of like man 
pages for Ruby libraries. You may access it like this:
  ri Classname
  ri Classname.class_method
  ri Classname#instance_method
If you do not wish to install this documentation in the future, use the
--no-document flag, or set it as the default in your ~/.gemrc file. See
'gem help env' for details.

RubyGems system software updated
Updating installed gems
Updating bundler
bundler's executable "bundle" conflicts with /Users/knutsen/.asdf/installs/ruby/2.5.1/bin/bundle
Overwrite the executable? [yN]  Successfully installed bundler-1.16.4
```

I answered "y", not sure if that was correct.

**Comment by knutsenm** [comment]
This might be related: https://github.com/asdf-vm/asdf-ruby/issues/47

**Comment by dknell** [comment]
This is not a problem with asdf. Looks like the issue is with the `mac` script trying to install the `bundler` gem when it was already installed. 

```
Updating bundler
bundler's executable "bundle" conflicts with /Users/dknell/.asdf/installs/ruby/2.6.1/bin/bundle
Overwrite the executable? [yN]  
```

The bundler gets installed via `install_asdf_language "ruby"`. Do we still need to run `gem_install_or_update "bundler"`? Seems like if we got rid of that line, this error would go away.

**Comment by croaky** [comment]
`gem_install_or_update "bundler"` can be removed because
as of Ruby 2.6, Bundler is now installed as a default gem.
https://www.ruby-lang.org/en/news/2018/12/25/ruby-2-6-0-released/

This is the only instance of `gem_install_or_update` used in the script,
so it could also be deprecated, perhaps like this?

Move the implementation below to
https://github.com/thoughtbot/laptop/wiki

```bash
gem_install_or_update() {                                                                                                                                  
  if gem list "$1" --installed > /dev/null; then                                                                                                           
    gem update "$@"                                                              
   else                                                                           
     gem install "$@"                                                                                                                                       
  fi                                                                                                                                                       
}  
```

Replace it with something like

```bash
gem_install_or_update() {
  echo "thoughtbot/laptop no longer provides gem_install_or_update"
  echo "see https://github.com/thoughtbot/laptop/wiki to copy an implementation into your laptop.local script if you need it"
  exit 1
}
```

In a few months, remove the error message.

The extra work is to help anyone who depends on `gem_install_or_update`
fail fast and migrate quickly.

**Comment by dknell** [comment]
@croaky I would think that many people are using `gem_install_or_update` in their `.laptop.local` scripts. Doesn't hurt to keep it there as a helper.

Thanks for confirming that the bundler gem is installed by default! I will submit a PR to remove that line.

**Comment by composerinteralia** [comment]
Closed with #557.
